src/prettytable/prettytable.py

```diff
@@ -1583,14 +1583,27 @@ def _compute_widths(self, rows, options):
                 title_width = 0
             min_table_width = self.min_table_width or 0
             min_width = max(title_width, min_table_width)
-            table_width = self._compute_table_width(options)
-            if table_width < min_width:
-                # To make enough space for the title, all columns have to
-                # grow enough that their combined widths (and the borders
-                # between them) are equal to the length of the title.
-                border_char_count = 3 * (len(widths) - 1)
-                scale = 1.0 * (len(options["title"]) - border_char_count) / sum(widths)
-                widths = [int(math.ceil(w * scale)) for w in widths]
+            if options["border"]:
+                borders = len(widths) + 1
+            else:
+                if options["preserve_internal_border"]:
+                    borders = len(widths) - 1
```

Comment:
Please could you add tests to cover this line?

Reply:
I have added tests to tables with only internal borders, I found out that it have to be `borders = len(widths)` actually, after this change all tests passed
When table is rendered with only internal borders, horizontal line is one character shorter, I have taken this into the account when writing test but I don't know if this is the correct behavior. Part of the code I have touched doesn't affect borders